etc/rc.conf

```diff
@@ -120,6 +120,8 @@
 # Or the ionice level. The format is class[:data] , just like the
 # --ionice start-stop-daemon parameter.
 #SSD_IONICELEVEL="2:2"
+# Or the OOM score adjustment.
+#SSD_OOMSCOREADJ="-1000"
```

Comment:
is `-1000` a good default to leave here ?  it basically means "never allow OOM to kill this".  if you want to keep that, it'd be worth a comment, and a ref to the appropriate man page for more details.

Reply:
I've used -100 for my own critical services. Would you find that a more palatable suggestion?

Reply:
i'm not saying `-1000` is never a good idea, but i strongly doubt that it should be the suggested default that people get by uncommenting a single line.  the number of services that you'd want using this on any given system would usually be extremely small (probably like <5).

plus, config files with commented out lines are supposed to represent the actual default rather than some other random value.  so here it'd be better to use `#SSD_OOMSCOREADJ="0"` (which i think is the default behavior, but i don't have a way of easily checking), and then leave a comment pointing people to more extensive documentation which would cover valid ranges, as well as the underlying meaning & implications of these settings, as well as some recommended conventions.

this kind of thing is why i suggested sep PR's for sep issues ;).

Reply:
> this kind of thing is why i suggested sep PR's for sep issues ;).

This kind of thing is why I put the bugfix first and the controversial stuff subsequent, so the bugfix could be merged while we continue to hash out the details of the enhancement. ;)

Reply:
I agree that the default should be the value in the commented-out example config option, but that's not the case for the other options currently. (The default nice level is not -19, and the default I/O nice is not 2:2.)

Reply:
we can/should fix the other bad ones ;).

i think there's a few reasonable methods for config files:
1. the commented value is the default
2. the inline docs clearly mention that the value is changing behavior from the default
3. provide both so the default is clear, and can list one or two common alternatives that users would want without having to refer to extensive manual pages

since our configs have largely been doing (1), we should favor consistency here and just switch them all to that.

Reply:
> > this kind of thing is why i suggested sep PR's for sep issues ;).
> 
> This kind of thing is why I put the bugfix first and the controversial stuff subsequent, so the bugfix could be merged while we continue to hash out the details of the enhancement. ;)

this is possible if doing manual git operations from the CLI, but afaik, not possible at all via the web interfaces.

i'm comfortable with manipulating/extracting git objects, but i don't think it's common, and even then, GH doesn't exactly make it an easy process.  figuring out the exact remote & ref to manually fetch is tedious.

a sep PR makes it a lot easier to just smash the merge button.  basically GH is tailored for flows where a single PR maps to a single "feature", and not tailored for unrelated things.  ignoring the "here's a grab bag of fixes that everyone agrees on" type of thing.

my main desktop had hardware failures recently, so i've (temporarily) lost easy access to do any manual surgery from the CLI.

Reply:
#436

Reply:
> 1. the commented value is the default
> 
> since our configs have largely been doing (1), we should favor consistency here and just switch them all to that.

The slight gotcha here is that setting `SSD_NICELEVEL="0"` will not actually cause `start-stop-daemon` and `supervise-daemon` to set the priority of the daemon process to 0. Rather, no change in priority will be applied. The same goes for the new `SSD_OOMSCOREADJ` variable: setting it to 0 is equivalent to leaving it unspecified: no change to the value will be applied. Granted, the expectation is that `start-stop-daemon` will be exec'd in a process that already has priority 0 and OOM score adjustment 0, but that's not necessarily the case. I'm not sure that the commented-out suggestions for these variables should have values that are no-ops.

Perhaps I ought to choose an out-of-range value as the sentinel ("unset") value so that explicitly specifying 0 for these options will cause the priority/oom_score_adj value to be affirmatively set to 0. Then the commented-out suggestions in the example config file could all meaningfully have 0 values.

Reply:
> Perhaps I ought to choose an out-of-range value as the sentinel ("unset") value so that explicitly specifying 0 for these options will cause the priority/oom_score_adj value to be affirmatively set to 0. Then the commented-out suggestions in the example config file could all meaningfully have 0 values.

3317c9bb92466735d247491a7848f2143e8c090d